### PR TITLE
Update resources for use with current Google provider, remove unused variables

### PR DESCRIPTION
**Pull request by SaraChandler**
In this PR: 

- Updates "google_project_iam_member" and "google_storage_bucket" resources for use with latest Google provider version
- Removes unused variables networking_firewall_ports and networking_ip_allow_list

### CODE REVIEW

File: modules/iam/README.md [hunk 1]
```diff
@@ -15,6 +16,7 @@ module "iam" {
 
   gcs_bucket_vault_license = "test-vault-license"
   key_ring_id              = "projects/project_id/locations/global/keyRings/test-vault-keyring"
+  project                  = "project_id"
```
Comment (troyready) [inline modules/iam/README.md:19]: Should be `project_id` to match the variable name

File: modules/iam/README.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 * `gcs_bucket_vault_license` - GCS bucket ID containing Vault license
 * `key_ring_id` - string value of the self link to the kms crypto key
+* `project` - string value of the GCP project in which to launch resources
```
Comment (troyready) [inline modules/iam/README.md:7]: Should be `project_id` to match the variable name

**Review by troyready — APPROVED** [review]
Looks good; cosmetic notes inline

File: modules/iam/README.md (hunk 1 above)
Comment (SaraChandler) [reply modules/iam/README.md]: Fixed [798d380](https://github.com/hashicorp/terraform-gcp-vault-ent-starter/pull/7/commits/798d3809ab24136aa8ba9945fc65df09a4d26876)

File: modules/iam/README.md (hunk 2 above)
Comment (SaraChandler) [reply modules/iam/README.md]: Fixed [e58f410](https://github.com/hashicorp/terraform-gcp-vault-ent-starter/pull/7/commits/e58f41061cdc0afe65896fa4029b4a715f3318ac)

**Pull request merged**